src/audio/dai.c

```diff
@@ -503,10 +503,8 @@ static int dai_prepare(struct comp_dev *dev)
 		return -EINVAL;
 	}
 
-	/* TODO: not sure what this wb is for? */
-	/* write back buffer contents from cache */
-	dcache_writeback_region(dd->dma_buffer->stream.addr,
-				dd->dma_buffer->stream.size);
+	/* clear dma buffer to avoid pop noise */
+	buffer_zero(dd->dma_buffer);
```

Comment:
does this also force the zeroed data to be written back?

Reply:
good question. This function does it.

````
static inline void buffer_zero(struct comp_buffer *buffer)
{
	tracev_buffer_with_ids(buffer, "stream_zero()");

	bzero(buffer->stream.addr, buffer->stream.size);
	if (buffer->caps & SOF_MEM_CAPS_DMA)
		dcache_writeback_region(buffer->stream.addr,
					buffer->stream.size);
}
````